pkg/cmd/server/origin/master_config.go

```diff
@@ -670,6 +671,21 @@ func newAuthenticator(config configapi.MasterConfig, restOptionsGetter restoptio
 		},
 	}
 
+	//	if we have a front proxy providing authentication configuration, wire it up and it should come first
+	if config.AuthConfig.RequestHeader != nil {
+		requestHeaderAuthenticator, err := headerrequest.NewSecure(
+			config.AuthConfig.RequestHeader.ClientCA,
+			config.AuthConfig.RequestHeader.ClientCommonNames,
+			config.AuthConfig.RequestHeader.UsernameHeaders,
+			config.AuthConfig.RequestHeader.GroupHeaders,
+			config.AuthConfig.RequestHeader.ExtraHeaderPrefixes,
+		)
+		if err != nil {
+			return nil, err
+		}
+		ret.Handlers = append([]authenticator.Request{requestHeaderAuthenticator}, ret.Handlers...)
```

Comment:
> I was going to make the front proxy tell us all the groups, but I don't feel strongly.

I changed my mind.  I think I feel strongly.  If the front proxy is telling us its `user=system:anonymous, group=system:unauthenticated`, then we don't want to add `system:authenticated`.